packages/flutter_tools/gradle/src/main/groovy/flutter.groovy

```diff
@@ -67,7 +66,7 @@ class FlutterExtension {
      * Specifies the relative directory to the Flutter project directory.
      * In an app project, this is ../.. since the app's Gradle build file is under android/app.
      */
-    String source
+    String source = "../.."
```

Comment:
Is there a test that covers when source is not set that we can still build?

Reply:
I'm was not sure if the test should be part of this change, as I don't propose to remove setting the source in `app/build.gradle`.

There are also two cases:
Having unset the source in `app/build.gradle` 

```
flutter {
    source ''
}
```

or having non source set at all in `app/build.gradle`

```
flutter {
}
```

https://github.com/flutter/flutter/blob/cbc18b2c51ca27b158976ff6cf9a429ff5bffa3f/packages/flutter_tools/templates/app_shared/android-java.tmpl/app/build.gradle.tmpl#L56

One can argue, that the build works without setting the source with this change and we should ensure it's working in the future, if that's what we actually want. I'll add a test and then can see if this fits the requirements.

Reply:
I think this change belongs in a separate PR

Reply:
I added a test file: android_gradle_source_path_test

Reply:
> I think this change belongs in a separate PR

Oh ok, my page didn't update to your respone. So I can add another PR just to change the default...

Reply:
Moved changes to #142934